src/helpers/WidgetTemplateExport.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+namespace hrzg\widget\helpers;
+
+use hrzg\widget\models\crud\base\WidgetTemplate;
+use yii\base\BaseObject;
+use yii\base\ErrorException;
+use yii\base\InvalidConfigException;
+use yii\helpers\FileHelper;
+use yii\helpers\Inflector;
+use yii\helpers\Json;
+
+/**
+ * --- PUBLIC PROPERTIES ---
+ *
+ * @property string $exportDirectory
+ * @property string $templateFilename
+ * @property string $schemaFilename
+ * @property-write  WidgetTemplate $widgetTemplate
+ * @property string $tarFileName
+ *
+ * --- MAGIC GETTERS ---
+ *
+ * @property-read string $tarFilePath
+ *
+ * @author Elias Luhr
+ */
+class WidgetTemplateExport extends BaseObject
+{
+    public const META_FILE = 'meta.json';
+    public const TEMPLATE_FILE = 'template.twig';
+    public const SCHEMA_FILE = 'schema.json';
+
+    /**
+     * Export directory of the generated tar file
+     * If this is set with an alias, it will be automatically resolved later on
+     *
+     * @var string
+     */
+    public $exportDirectory = '@runtime/tmp/dmstr/widgets/templates';
```

Comment:
directories (and/or filenames) used for such exports should always have some randomness to prevent race conditions.

Reply:
This is now a protected property and is set to a random directory in /tmp (via [sys_get_temp_dir](https://www.php.net/manual/de/function.sys-get-temp-dir.php))